src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1729,6 +1729,81 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
                 return;
             }
 
+            var usage = ticket.GetUsage();
+            var identifier = ticket.GetProperty(OpenIdConnectConstants.Extra.ClientId);
+            var audiences = ticket.GetAudiences();
+            if (string.Equals(usage, OpenIdConnectConstants.Usages.RefreshToken, StringComparison.Ordinal)) {
+                // For Refresh tokens
+                if ((clientNotification.IsSkipped &&
+                     ticket.ContainsProperty(OpenIdConnectConstants.Extra.ClientAuthenticated)) ||
+                    (!clientNotification.IsSkipped &&
+                     (!ticket.ContainsProperty(OpenIdConnectConstants.Extra.ClientAuthenticated) ||
```

Comment:
I'm not sure it's really interesting to prevent authenticated callers from validating "unauthenticated" tickets from a security perspective, since the caller can simply stop sending its credentials to work around this measure.